treescript/src/treescript/l10n.py

```diff
@@ -77,26 +81,82 @@ def build_platform_dict(bump_config, repo_path):
     return platform_dict
 
 
+# get_latest_revision {{{1
+async def get_latest_revision(locale, url):
+    """Download the hg pushlog for the latest locale revision.
+
+    Args:
+        locale (str): the locale to query
+        url (str): the [templatized] pushlog url
+
+    Returns:
+        tuple (locale, revision)
+
+    """
+    url = url % {"locale": locale}
+    with tempfile.NamedTemporaryFile() as fp:
```

Comment:
nit: is there a reason this has to go to disk instead of just keeping it in memory? size?

Reply:
> This looks fine. Do we need any patches to central, beta, or release to pin some locales before this lands?

Thanks! No, pinning will likely be manual once we determine the default revision for a locale is bad, and we're unable to fix the locale sooner.

Reply:
No, no reason.

Reply:
Hm, I was going to port over `scriptworker.util.load_json_or_yaml_from_url` but that downloads the file and loads it from a tempfile as well. I'll probably leave this for another patch if we want to optimize.